homeassistant/helpers/bluetooth.py

```diff
@@ -0,0 +1,90 @@
+"""Common functions related to Bluetooth device management."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from homeassistant.components.bluetooth.passive_update_processor import (
+    PassiveBluetoothDataUpdate,
+    PassiveBluetoothEntityKey,
+)
+from homeassistant.components.sensor import SensorEntityDescription
+
+from .sensor import sensor_description_to_key, sensor_device_info_to_hass_device_info
+
+if TYPE_CHECKING:
+    # `sensor_state_data` is a second-party library (i.e. maintained by Home Assistant
+    # core members) which is not strictly required by Home Assistant.
+    # Therefore, we import it as a type hint only.
+    from sensor_state_data import (
```

Comment:
I think that this comment says it all and why we should probably not do it.

Reply:
agreed.

Reply:
For context, this admonition (copypasted here into the bt-specific file) was introduced in https://github.com/home-assistant/core/pull/81905 per request.

Reply:
Then we should revert that PR too because it elevates a dependency to core-dependency status (required for type checking core).

Reply:
I was told in https://github.com/home-assistant/core/pull/81790#discussion_r1017657620 that the expectation is for all deps, not just core deps, to be installed when running tests.